Source/ProgressionSystemRuntime/Private/Widgets/PSEndGameWidget.cpp

```diff
@@ -160,3 +165,28 @@ void UPSEndGameWidget::UpdateStarProgressBarValue(const FPoolObjectData& Created
 	UPSStarWidget& SpawnedWidget = CreatedData.GetChecked<UPSStarWidget>();
 	SpawnedWidget.UpdateProgressionBarPercentage(NewProgressBarValue);
 }
+
+// Updates the progression menu widget when player changed
+void UPSEndGameWidget::UpdateProgressionWidgetForPlayer()
+{
+	UPSSaveGameData* SaveGameData = UPSWorldSubsystem::Get().GetCurrentSaveGameData();
```

Comment:
this condition looks suspicious next reasons:
- `SaveGameData` in this function at all
- if its added to filter updation when save is not created, then it's hack, as `OnScoreChanged` delegate itself shouldnt even call when save is not created. No scores were changed. For this case, systems should bind to `OnInitialize` delegate to perform initial update

Please reconsider it and remove the condition

Reply:
removed: https://github.com/h4rdmol/ProgressionSystem/pull/45/commits/0279093f34c6ca88b8ced79f917e8a4fabf363c8